src/Core/Domain/Country/QueryResult/Countries.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Country\QueryResult;
+
+/**
+ * Holds list of countries found by query configuration
+ */
+class Countries
+{
+    /**
+     * @var array
+     */
+    private $countries;
+
+    /**
+     * @param array $countries
+     */
+    public function __construct(array $countries)
+    {
+        $this->countries = $countries;
+    }
+
+    /**
+     * @return array
+     */
+    public function getCountries(): array
+    {
+        return $this->countries;
```

Comment:
Isn't it legacy array, but wrapped into `Countries`? I wonder if we can do better. :wink: 

@matks wdyt?

Reply:
Yes, we should create a VO Country and this would be a collection of Country VO

Reply:
created a VO Country and refactored this to be a collection of Country VO